### PR TITLE
Add all pairs longest paths utility

### PR DESCRIPTION
**Pull request by harsh-nod**
This PR adds a function that modifies Floyd-Warshall to compute the longest paths in a graph with
cycles with the assumption that the cycles don't
have positive weights.

### CODE REVIEW

File: shark_turbine/kernel/wave/scheduling/graph_utils.py [hunk 1]
```diff
@@ -136,3 +153,40 @@ def find_cycles_in_graph(
...
+def all_pairs_longest_paths(
+    graph: fx.Graph,
+    edges: list[Edge],
+) -> dict[tuple[int, int], IndexExpr]:
+    """
+    For each node in the graph, compute the longest path to all other nodes.
+    Uses the Floyd-Warshall algorithm and assumes that the cycles don't
+    have positive weights.
+    """
+    T = index_symbol("$INITIATION_INTERVAL")
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/scheduling/graph_utils.py:167]: Move it to global var?

File: shark_turbine/kernel/wave/scheduling/graph_utils.py [hunk 2]
```diff
@@ -136,3 +153,40 @@ def find_cycles_in_graph(
...
+        for w in graph.nodes:
+            D[(v, w)] = -math.inf
+    for edge in edges:
+        D[(edge._from, edge._to)] = (
+            edge.weight.delay - edge.weight.iteration_difference * T
+        )
+    for u in graph.nodes:
+        for v in graph.nodes:
+            for w in graph.nodes:
+                D[(v, w)] = sympy.Max(D[(v, w)], D[(v, u)] + D[(u, w)])
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/scheduling/graph_utils.py:179]: N^3 complexity is unfortunate, but we can live with it for, I suppose.

**Review by Hardcode84 — APPROVED** [review]

File: shark_turbine/kernel/wave/scheduling/graph_utils.py (hunk 1 above)
Comment (harsh-nod) [reply shark_turbine/kernel/wave/scheduling/graph_utils.py:167]: sure will do it in the next PR.

File: shark_turbine/kernel/wave/scheduling/graph_utils.py (hunk 2 above)
Comment (harsh-nod) [reply shark_turbine/kernel/wave/scheduling/graph_utils.py:179]: yeah I dont think there's a way around this unfortunately.

**Pull request merged**